### ISSUE TITLE
Improving navigation for macOS with modifier keys

### ISSUE THREAD

**Issue by Spaxe** [damianavila/RISE#278]
Ref #180 

I'd like to raise a usability issue, at least for people on MacOS (bear with me).

To access PgUp/PgDn, on the MacBooks, you press Fn + Up / Fn + Down. (I assume this is also the case for some others.)

However, Fn + Left / Fn + Right emits Home / End, which skips the slides to the beginning or end.

So you have this disparity in controls, when you hold down Fn key, the behaviours in arrows are very different.

Here's a proposal:

- Alt + Left: Previous Slide
- Alt + Right: Next Slide
- Alt + Up: Previous Sub-Slide
- Alt + Down: Next Sub-Slide 

Essentially - the control would be the same as Reveal.js, but with a modifier. The experience will be a lot smoother for the presenters. No more Fn key confusion.

Open to suggestions, and it will be worth doing some more research into how people are making their own controls. Also, I don't want to accidentally switch workspaces on operating systems, so the hotkeys will need some closer examination.

**Comment by standage** [comment]
I agree, the inconsistency was jarring at first. I can navigate right and left with just the left and right arrows, which is intuitive. But it took me a while to figure out that Fn was needed to make the up and down arrows work with subslides.

**Comment by damianavila** [comment]
> I agree, the inconsistency was jarring at first

I know... I develop on Linux so this is probably a bias from my side :wink:
We should improve this, certainly.

**Comment by stsievert** [comment]
I ran into this issue, and it bothered me enough to find this. I only learned about the fn keys from this issue.

**Comment by larsgeorge** [comment]
Upvoting, since on a Mac this is otherwise highly confusing. Bump.

**Comment by parmentelat** [comment]
I'd like to reiterate my comment on using spacebar and Shift-spacebar, which is **way** more convenient than navigating right/down - it frees your brain from having to compute which direction is next

that being said, the reason why we're not able to react faster on such requests is because of issue #442; right now we basically defer keyboard management to reveal.js itself; if #442 was taken care of you/we would have much greater flexibility in shortcut assignment

but I can't give any timeframe for that to happen, so in the meanwhile please do me a favour and give this space/shift-space thing a try :)

**Comment by damianavila** [comment]
@parmentelat I have seen a lot of people using the directional keys instead of the shift/shift-space ones. Maybe we should make this more **strongly** represented (discoverable) in the docs. I am pretty sure people will find it really useful, but they are not getting that info, even when we mention it in the docs, IMHO.

**Comment by parmentelat** [comment]
yeah that's my feeling too; that's why I wrote examples/README in the first place :)

another option would be to rebind the arrow keys; of course this is likely to be intrusive; it would be less so if we had #442, in the sense that people truly motivated to preserve the old behaviour could do so, while with the current code we can only hard-wire that

what else would you suggest / have in mind for the near term ?

**Comment by damianavila** [comment]
> another option would be to rebind the arrow keys; of course this is likely to be intrusive;

Yep, too intrusive...

> what else would you suggest / have in mind for the near term ?

I am just thinking of a big note in the docs, in the corresponding section for now.
